pkg/plugin/datasource.go

```
@@ -119,75 +119,112 @@ func (d *Datasource) query(_ context.Context, pCtx backend.PluginContext, query
 		- Nested List: [  [ {...}, {...}, ... ], {...}, ... ]
 	*/
 
-	//Recup GTSList of the response body
-	var gtsList b.GTSList
-
 	// If the response is a table...
+	backendTableResult, err := parseTableResult(body)
```

Not a huge fan of this way of handling different kind of available result. 

For example if the line https://github.com/CleverCloud/clevercloud-warp10-datasource/pull/5/files#diff-213832a772024707da1a5234b077c6f336e32d7316a0bfb5d228517e9886febeR199 return an error we loose the level of information and we still compute other kind of possible output. 

I would have split into two steps, once method to check if the type is a `TableResult`, then I would have extract the `TableResult` in a second method

---

> I would have split into two steps, once method to check if the type is a TableResult, then I would have extract the TableResult in a second method

yes it would be a safer approach to avoid masking errors